sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -900,12 +907,54 @@ class MicroBatchExecution(
    */
   protected def markMicroBatchExecutionStart(execCtx: MicroBatchExecutionContext): Unit = {}
 
+  private def updateCheckpointIdForOperator(
+      execCtx: MicroBatchExecutionContext,
+      opId: Long,
+      checkpointInfo: Array[StateStoreCheckpointInfo]): Unit = {
+    // TODO validate baseCheckpointId
+    checkpointInfo.map(_.batchVersion).foreach { v =>
+      assert(
+        execCtx.batchId == -1 || v == execCtx.batchId + 1,
+        s"version $v doesn't match current Batch ID ${execCtx.batchId}")
+    }
+    currentCheckpointUniqueId.put(opId, checkpointInfo.map { c =>
+      assert(c.checkpointId.isDefined)
+      c.checkpointId.get
+    })
+  }
+
+  private def updateCheckpointId(
```

Comment:
Let me make sure I understand the flow here:

1. Micro-batch ends, we call `updateCheckpointId`
2. This goes through every stateful operator and calls `updateCheckpointIdForOperator`
3. For each operator, we call into its `getCheckpointInfo` method
    1. That method will access the `checkpointInfoAccumulator`
    2. The `checkpointInfoAccumulator` is appended to using the unique ID from the state store after processing all data on the task
 4. In the future, we'll write this to the commit log.

Is this right?